armi/reactor/blueprints/__init__.py

```diff
@@ -326,7 +326,8 @@ def _prepConstruction(self, cs):
                     for a in list(self.assemblies.values())
                     if not any(a.hasFlags(f) for f in assemsToSkip)
                 )
-                axialExpansionChanger.axialExpansionChanger.expandColdDimsToHot(
+                chngrT = getPluginManager().hook.getAxialExpansionChanger()
```

Comment:
This will likely expose my lack of understanding about the inner workings of the plugins systems, but what happens if an Application (a "Plugin" in ARMI terms) has Plugins (also "Plugin" in ARMI terms) that have multiple axialExpansionChanger subclasses? How would those get resolved? 

E.g.: 
ARMI_application:
- Plugin A
- Plugin B (has axialExpansionChangerForPluginB)
- Plugin C
- Plugin D (has axialExpansionChangerForPluginD)

Reply:
Good question. The call order for plugins is first in first out. So PluginB would have its class returned first.

My though process was that ARMI would provide a sensible default expander, and then a single plugin for an application would provide that application's expander. You could also envision a world where an application brings in an `AxialExpansionInterface` (e.g., for things like `interactEveryNode` through an `AxialExpansionPlugin` and that plugin would provide this hook as well.

Reply:
@albeanth added this to the hook docstring

        This hook should only be provided by one additional plugin in your application. Otherwise
        the `order of hook execution <https://pluggy.readthedocs.io/en/stable/index.html#call-time-order>`_
        may not provide the behavior you expect.

https://github.com/terrapower/armi/pull/1870/commits/9dbbca288fa024d87da69bd5731b58045ad9d174